account_payment_order/models/account_payment_order.py

```diff
@@ -246,7 +246,7 @@ def create(self, vals):
     @api.onchange("payment_mode_id")
     def payment_mode_id_change(self):
         if len(self.allowed_journal_ids) == 1:
-            self.journal_id = self.allowed_journal_ids
+            self.journal_id = self.allowed_journal_ids._origin
```

Comment:
Is this not working?

```suggestion
            self.journal_id = self.allowed_journal_ids.id
```

My doubt about your solution is what happens if you change the payment mode to another.